### DISCUSSION TITLE
多维数组如何初始化

### DISCUSSION THREAD

**Discussion by TommyPlayer-c** [arcsysu/SYsU-lang#40] [Q&A] [answered]
```C++
int main(){
    int a[4][2]={1,2,3,4,5,6,7,8};
    int b[4][2]={{a[0][0],a[0][1]},{3,4},{5,6},{7,8}};
    return 0;
}
```
函数体内定义了局部二维数组，看不懂clang生成的ir代码逻辑。是要使用 memcpy 来初始化吗，但这个指令我不太会使用，助教能不能给出具体的使用方法？或者是使用别的方法来初始化多维数组？

**Comment by wu-kan** [comment, marked as answer]
c语言中多维数组的本质是一维数组？

**Reply by TommyPlayer-c** [reply]
仔细看了看 IR 生成的代码，好像是一个元素一个元素的赋值，要熟悉 GEP 指令，https://zhuanlan.zhihu.com/p/66909226

**Reply by paprika0741** [reply]
> c语言中多维数组的本质是一维数组？
全局数组如何使用GEP呢

**Reply by Yorkking** [reply]
> > c语言中多维数组的本质是一维数组？
> > 全局数组如何使用GEP呢

本质当然是本质，计算机本质上还是个图灵机呢。

**Reply by Yorkking** [reply]
> > c语言中多维数组的本质是一维数组？
> > 全局数组如何使用GEP呢

好像有一个 llvm::ConstantArray 的API，可以研究一下？当然，也可以直接粗暴地转为局部数组进行初始化。
